kong/db/schema/typedefs.lua

```diff
@@ -214,7 +214,7 @@ typedefs.header_name = Schema.define {
 
 typedefs.timeout = Schema.define {
   type = "integer",
-  between = { 0, math.pow(2, 31) - 2 },
+  between = { 1, math.pow(2, 31) - 2 },
```

Comment:
This is considered part of the public API of 1.0 (even though unfortunately not documented yet). We may not know, but third-party plugins could be relying on this behaviour. Some of our external plugins (bundled at packaging time) may also make use of this typedef (worth checking...).

Most likely, as you were saying, a value of `0` for a timeout is acceptable. As such, why not making that the general use-case (and keep `0` here, not only for backwards-compatibility, but also commodity), and update the timeout fields of the Service entity to _not_ accept `0`?

Reply:
> Some of our external plugins (bundled at packaging time) may also make use of this typedef (worth checking...).

I checked, none are.

> Most likely, as you were saying, a value of 0 for a timeout is acceptable. As such, why not making that the general use-case (and keep 0 here, not only for backwards-compatibility, but also commodity), and update the timeout fields of the Service entity to not accept 0?

I would accept the PR as is, but I'm fine with this option too.

Reply:
I don't see why we wouldn't follow the above approach. Timeouts in general can have values of `0`, it just so happens that the timeout attributes of this particular Service entity cannot. Plus as I said, we preserve backwards-compatibility for any use-case out there we may not be aware of.

Reply:
I've updated the PR to avoid the breaking change. Please re-review, thanks!